gtsam_unstable/slam/ProjectionFactorRollingShutter.h

```diff
@@ -0,0 +1,257 @@
+/* ----------------------------------------------------------------------------
+
+ * GTSAM Copyright 2010, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * Authors: Frank Dellaert, et al. (see THANKS for the full author list)
+
+ * See LICENSE for the license information
+
+ * -------------------------------------------------------------------------- */
+
+/**
+ * @file ProjectionFactorRollingShutter.h
+ * @brief Basic projection factor for rolling shutter cameras
+ * @author Yotam Stern
+ */
+
+#pragma once
+
+#include <gtsam/nonlinear/NonlinearFactor.h>
+#include <gtsam/geometry/PinholeCamera.h>
+#include <gtsam/geometry/CalibratedCamera.h>
+#include <gtsam/geometry/Cal3_S2.h>
+#include <boost/optional.hpp>
+
+namespace gtsam {
+
+/**
+ * Non-linear factor for 2D projection measurement obtained using a rolling shutter camera. The calibration is known here.
```

Comment:
Did you format all new code with Google style? Suspiciously long lines here.

Reply:
I did - comments still look quite long though

Reply:
I reformatted with Google style and lots changed. Check your IDE :-)

Reply:
<img width="611" alt="Screen Shot 2021-08-28 at 5 37 38 PM" src="https://user-images.githubusercontent.com/4977344/131231627-a2d85edd-1489-44f4-89f1-9239f48b70e6.png">

what do you use on your side?

Reply:
clang-format, Google style.